src/openscm_units/_unit_registry.py

```diff
@@ -247,6 +275,15 @@ class ScmUnitRegistry(pint.UnitRegistry):
 
     _contexts_loaded = False
 
+    def __init__(self):
+        pint.UnitRegistry.__init__(self)
+
+        with open(
```

Comment:
I added a commit moving the actual loading to add_standards(). However, that's not really lazy because it is still done at import time automatically. Since the mixtures define not only conversions, but also units, I think loading them can't be done lazily (at least with the current infrastructure). If loading of a yaml file at import time should be avoided at all cost, I could maybe convert the yaml file to python or pon or include part of the data directly into _unit_registry.py like it is done for the `_STANDARD_GASES`, but for me, that seems more confusing during maintenance compared to a single yaml file.

Reply:
All very true. In my head, I had imagined not loading the mixture gases at all until requested (i.e. you'd have to do `registry.load_mixtures()` before having access to them, similar to how `_load_metric_conversions` (https://github.com/openscm/openscm-units/blob/8f11d22dccbf73b06d17200161b311c496b46f72/src/openscm_units/_unit_registry.py#L345) has to be called before you can actually access the metric conversions).

If having to remember to call e.g. `load_mixtures` sounds like more hassle than it's worth, we can also leave with the file reading.

Reply:
I've done some measurements:
```python
In [1]: import yaml

In [2]: def load_mixtures():
   ...:     with open('mixtures.yaml') as fd:
   ...:         a = yaml.safe_load(fd)
   ...: 

In [3]: %time load_mixtures()
CPU times: user 94 ms, sys: 101 µs, total: 94.1 ms
Wall time: 92.9 ms

In [4]: !sudo sync

In [5]: !sudo sh -c 'echo 3 > /proc/sys/vm/drop_caches'

In [6]: %time load_mixtures()
CPU times: user 90.8 ms, sys: 4.17 ms, total: 95 ms
Wall time: 94.8 ms

In [7]: %time import openscm_units
CPU times: user 860 ms, sys: 767 ms, total: 1.63 s
Wall time: 984 ms
```
So, loading the mixtures would add about 10% to the total import time.

If I convert the yaml file to a python file instead with a straight dict literal and `import` that instead of `yaml.safe_load`ing it, the loading time of the mixtures is greatly reduced:
```python
In [1]: !sudo sync

In [2]: !sudo sh -c 'echo 3 > /proc/sys/vm/drop_caches'

In [3]: %time from mixtures import mixtures
CPU times: user 2.09 ms, sys: 0 ns, total: 2.09 ms
Wall time: 2.86 ms
```
The "0 ns" system time here is suspicious, but I guess my machine caches the file on a lower level (the SSD or such). However, since the parsing (user time) was the main thing also for the yaml file, at least the order of magnitude of the speed-up (about 50 times faster) should be right.

While on it, I also profiled importing openscm-units more generally, and found that when importing and loading conversions, about 80% of the time is spent importing (importing pint takes at least half the time) and 20% of the time is spent loading the conversions on first use. From the time it takes to load conversions, roughly 40% (so 8% of the total) is spent reading the CSV and 50% (10% of the total) is spent adding the transformations to the contexts.

I'll put the mixtures into a python file so that we don't spend time parsing yaml files on each import (after all, a python literal is about as readable as a yaml file anyway). That way we don't really degrade import performance by much (less than 1%) and everything is usable right after import.

Reply:
Super nice analysis! Thanks heaps @mikapfl